### PR TITLE
Add output exports needed by ch01s04

### PR DESCRIPTION
**Pull request by niroy**

### CODE REVIEW

File: aws/cloudformation/privatelink-with-network-firewall.yaml [hunk 1]
```diff
@@ -744,6 +744,8 @@ Outputs:
       ROSA private VPC CIDR block. Use that CIDR block for the --machine-cidr
       option of the "rosa create cluster" command.
     Value: !GetAtt rRosaVpc.CidrBlock
+    Export:
+      Name: RosaVpcCidr
```
Comment (herve4m) [inline aws/cloudformation/privatelink-with-network-firewall.yaml:748]: In real life, we should ensure that exports are unique in the account/region. It is a good practice to prefix them with the stack name (if the customer deploys several clusters for example).
Anyway, you can keep it this way. We'll see if it raises issues.

```suggestion
      Name: !Join [":", [!Ref "AWS::StackName", RosaVpcCidr]]
```

File: aws/cloudformation/privatelink-with-network-firewall.yaml [hunk 2]
```diff
@@ -831,10 +833,14 @@ Outputs:
     Description: Transit gateway that connects the two VPCs
     Value:
       Ref: rTransitGateway
+    Export:
+      Name: TransitGateway
```
Comment (herve4m) [inline aws/cloudformation/privatelink-with-network-firewall.yaml:837]:
```suggestion
      Name: !Join [":", [!Ref "AWS::StackName", TransitGateway]]
```

File: aws/cloudformation/privatelink-with-network-firewall.yaml [hunk 3]
```diff
@@ -831,10 +833,14 @@ Outputs:
...
   oPrivateTransitGatewayAttachment:
     Description: Transit gateway attachment to the ROSA private VPC
     Value:
       Ref: rRosaTransitGatewayAttachment
+    Export:
+      Name: RosaTransitGatewayAttachment
```
Comment (herve4m) [inline aws/cloudformation/privatelink-with-network-firewall.yaml:843]:
```suggestion
      Name: !Join [":", [!Ref "AWS::StackName", RosaTransitGatewayAttachment]]
```

File: aws/cloudformation/privatelink-with-network-firewall.yaml [hunk 4]
```diff
@@ -851,6 +857,8 @@ Outputs:
     Description: Transit gateway route table for the ROSA private VPC
     Value:
       Ref: rTransitGatewayRosaRouteTable
+    Export:
+      Name: TransitGatewayRosaRouteTable
```
Comment (herve4m) [inline aws/cloudformation/privatelink-with-network-firewall.yaml:861]:
```suggestion
      Name: !Join [":", [!Ref "AWS::StackName", TransitGatewayRosaRouteTable]]
```

**Review by herve4m — APPROVED** [review]
See my comments. You do not have to fix them, it's up to you.

File: aws/cloudformation/privatelink-with-network-firewall.yaml (hunk 1 above)
Comment (niroy) [reply aws/cloudformation/privatelink-with-network-firewall.yaml:748]: I understand that, but that means we'll have to hardcode the stack name is the cloudformation template as ROSA-infra, so I avoided that.

**Pull request merged**